library/src/actions/jwt/jwt.ts

```diff
@@ -0,0 +1,193 @@
+import type {
+  BaseIssue,
+  BaseValidation,
+  ErrorMessage,
+} from '../../types/index.ts';
+import { _addIssue } from '../../utils/index.ts';
+
+const POSSIBLE_JWT_REGEX = /^(?:[\w-]+\.){2}[\w-]+$/u;
+
+// https://datatracker.ietf.org/doc/html/rfc7518#section-3.1
+type JwtAlgorithm =
+  | 'HS256'
+  | 'HS384'
+  | 'HS512'
+  | 'RS256'
+  | 'RS384'
+  | 'RS512'
+  | 'ES256'
+  | 'ES384'
+  | 'ES512'
+  | 'PS256'
+  | 'PS384'
+  | 'PS512'
+  | 'none';
+
+function isValidJwt(
+  input: string,
+  jwtAlgorithm: JwtAlgorithm | undefined
+): boolean {
+  if (!POSSIBLE_JWT_REGEX.test(input)) {
+    return false;
+  }
+  try {
+    // `atob`
+    // - for node env, marked as legacy
+    //     https://github.com/DefinitelyTyped/DefinitelyTyped/issues/65494
+    // - assumes the passed input to be base64 encoded,
+    //     will have to convert base64Url encoded header to base64
+    const header = JSON.parse(atob(input.split('.')[0]));
+    return (
+      'typ' in header &&
+      header.type === 'JWT' &&
+      (!('alg' in header) || header.alg === (jwtAlgorithm ?? 'none'))
+    );
+  } catch {
+    return false;
+  }
+}
+
+/**
+ * Jwt issue type.
+ */
+export interface JwtIssue<
+  TInput extends string,
+  TJwtAlgorithm extends JwtAlgorithm | undefined,
+> extends BaseIssue<TInput> {
+  /**
+   * The issue kind.
+   */
+  readonly kind: 'validation';
+  /**
+   * The issue type.
+   */
+  readonly type: 'jwt';
+  /**
+   * The expected property.
+   */
+  readonly expected: null;
+  /**
+   * The received property.
+   */
+  readonly received: `"${string}"`;
+  /**
+   * The algorithm used to sign the jwt.
+   */
+  readonly algorithm: TJwtAlgorithm;
```

Comment:
For some actions, there is a `requirement` property. Initially I concluded, the algorithm can be considered as a requirement but after thinking about it, I decided to add a new key because I think a `requirement` is something complete. Example: the `length` action with requirement set to `5` would mean testing the input with the requirement, nothing else. But in this case there are two steps. So, should I use the `requirement` key? If yes, what value should I set it to? What was the purpose of the `requirement` key in the issue and action objects?

Reply:
I have answered this at the other comment.